pkg/agent/openflow/pipeline.go

```diff
@@ -687,15 +689,22 @@ func (c *client) serviceLBBypassFlow(ipProtocol binding.Protocol) binding.Flow {
 }
 
 // l2ForwardCalcFlow generates the flow that matches dst MAC and loads ofPort to reg.
-func (c *client) l2ForwardCalcFlow(dstMAC net.HardwareAddr, ofPort uint32, category cookie.Category) binding.Flow {
+func (c *client) l2ForwardCalcFlow(dstMAC net.HardwareAddr, ofPort uint32, skipIngressRules bool, category cookie.Category) binding.Flow {
 	l2FwdCalcTable := c.pipeline[l2ForwardingCalcTable]
+	nextTable := ingressEntryTable
+	if skipIngressRules {
```

Comment:
I am thinking is it possible to leverage the "ofport" value to decide next table? e.g., if the value is tunnel port or gateway port (I assume we have no ingress rules applied on the gw0), then skipIngressRules, otherwise go to ingressEntryTable.
Then we could not use the parameter "skipIngressRule"

Reply:
Yes, that is possible too, but do you think better to put the logic in l2FwdCalcTable and pipeline.go or the callers?

Reply:
Personnally, I would prefer to put it inside the l2FwdCalcTable's logic, that would make the call simple, and easier to understand. If in the callers, reader should learn the pipeline case by case.

Reply:
Ok. Changed as your suggested.